src/rpcmining.cpp

```diff
@@ -211,7 +211,7 @@ UniValue setgenerate(const UniValue& params, bool fHelp)
     if (params.size() > 0)
         fGenerate = params[0].get_bool();
 
-    int nGenProcLimit = -1;
+    int nGenProcLimit = GetArg("-genproclimit", DEFAULT_GENERATE_THREADS);
```

Comment:
Yes, it fixes one bug:  If genproclimit is omitted to RPC setgenerate, don't change it; (copied from commit msg)